File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       unordered_map<int,int> total;
       int n=nums.size();
       for(int i=0;i<n;i++){
        int diff=target-nums[i];
        if(total.find(diff) != total.end()){
            return {total[diff],i};
        }
        else{
            total.insert({nums[i],i});
        }
       }
       return {};
    }
};

File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        mid=len(nums)//2
        print(mid)
        if(target>nums[mid]):
            j=mid
            while j<len(nums):
                if nums[j]==target:
                    return j
                j+=1
                
        else:
            j=0
            while j<=mid:
                if nums[j]==target:
                    return j
                j+=1
        return -1